frontend/src/app/search.tsx

```diff
@@ -183,36 +203,33 @@ export default class Search extends Component {
     console.log('handleSelectAll')
     this.setState({check: !this.state.check}, () => {
       this.setState(prevState => {
+        this.state.allPaths=[]
+        this.transientPaths=[]
         const selectAllcheck = this.state.data.map(dataitem => {
               dataitem["checkedItem"] = this.state.check
               console.log(dataitem["pant:transientPath"]+":"+dataitem["checkedItem"])
-              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):delete this.state.allPaths[this.state.allPaths.indexOf(dataitem["pant:transientPath"])]
+              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):this.state.allPaths.splice(this.state.allPaths.indexOf(dataitem["pant:transientPath"]),1)
```

Comment:
It's hard for me to read what's going on in this line - it looks like if `this.state.check` is true, then we push the transientPath into allPaths. Makes sense. But if `this.state.check` is false, then we splice a value out of allPaths... except we just set `this.state.allPaths` to `[]` 5 lines above, didn't we? So is this really necessary?

Reply:
The splice is required if we an uncheck the select all box.
But, the below 2 lines of code-
        this.state.allPaths=[]
        this.transientPaths=[]
will clear the transientPaths[] and allPaths[] when we try to use the select all feature for the second time in a row.
If we dont do that, the transientPaths[] and allPaths[], will carry forward the values which have been deleted, and in the next delete operations, these values are not found.
Hence, they need to be cleared.

Reply:
This has been taken care of after the discussion we had on this.